src/Tests/Microsoft.NET.Sdk.WorkloadManifestReader.Tests/SdkDirectoryWorkloadManifestProviderTests.cs

```diff
@@ -1243,7 +1274,99 @@ var sdkDirectoryWorkloadManifestProvider
             GetManifestContents(sdkDirectoryWorkloadManifestProvider)
                 .Should()
                 .BeEquivalentTo("iOS: iOS-6.0.100");
+        }
+
+        [Fact]
+        public void WorkloadResolverUsesManifestsFromWorkloadSet()
+        {
+            Initialize("8.0.200");
+
+            string manifestContents1 = """
+    {
+        "version": "11.0.1",
+        "workloads": {
+            "ios": {
+                "description": "iOS workload",
+                "kind": "dev",
+                "packs": [ "Microsoft.NET.iOS.Workload" ]
+            },
+        },
+        "packs": {
+            "Microsoft.NET.iOS.Workload" : {
+                "kind": "sdk",
+                "version": "1"
+            }
+        }
+    }
+    """;
+
+            string manifestContents2 = """
+    {
+        "version": "11.0.2",
+        "workloads": {
+            "ios": {
+                "description": "iOS workload",
+                "kind": "dev",
+                "packs": [ "Microsoft.NET.iOS.Workload" ]
+            },
+        },
+        "packs": {
+            "Microsoft.NET.iOS.Workload" : {
+                "kind": "sdk",
+                "version": "2"
+            }
+        }
+    }
+    """;
+
+            string manifestContents3 = """
+    {
+        "version": "12.0.1",
```

Comment:
Do we have any tests covering manifest versions that are prereleases and/or have full buildmetadata, e.g. "1.0.0-preview.1.23361.1+c291ddefb05ff3dd6a0b51df152bba794908dc64"?

Reply:
We have various tests that cover prerelease versions.  We don't have any with build metadata.  Is there anything specific with build metadata you think we should be testing?  I don't think we'll be using it in the versions we use for workloads.

Reply:
No, if we have ones with prerelease parts then that's fine, I couldn't recall if we did.